tests/rptest/tests/pandaproxy_test.py

```diff
@@ -2235,3 +2236,78 @@ def do_coordinator_change(utils: GroupCoordinatorTransferUtils):
         self.logger.debug(
             "Test offset commit: after second coordinator change")
         do_consumer_offset_commit(new_offset=1)
+
+
+class PandaProxyCompressedBatchesTest(PandaProxyEndpoints):
+    topics = [
+        TopicSpec(),
+    ]
+
+    def __init__(self, context):
+        super(PandaProxyCompressedBatchesTest,
+              self).__init__(context,
+                             extra_rp_conf={
+                                 "enable_leader_balancer": False,
+                                 "group_topic_partitions": 1
+                             })
+
+    @cluster(num_nodes=4)
+    def test_fetch(self):
+        producer = RpkProducer(self.test_context,
+                               self.redpanda,
+                               self.topic,
+                               msg_size=1024,
+                               msg_count=1024,
+                               printable=True,
+                               compression_type="snappy")
+        producer.start()
+        producer.wait()
+
+        self.logger.info("Fetching compressed records from partition 0")
+        fetch_result_raw = self._fetch_topic(self.topic, partition=0)
+        assert fetch_result_raw.status_code == requests.codes.ok
+        result = fetch_result_raw.json()
+        self.logger.debug(f"fetch result {json.dumps(result)}")
+        for item in result:
+            assert item["topic"] == self.topic
+            assert len(item["key"]) > producer.approx_key_size
+            assert len(item["value"]) > producer.approx_msg_size
+            assert item["partition"] == 0
+            assert item["offset"] > -1
+
+    @cluster(num_nodes=4)
+    def test_fetch_consumer_group(self):
+        producer = RpkProducer(self.test_context,
+                               self.redpanda,
+                               self.topic,
+                               msg_size=1024,
+                               msg_count=1024,
+                               printable=True,
+                               compression_type="snappy")
+        producer.start()
+        producer.wait()
+
+        self.logger.info("Create a consumer group")
+        group_id = f"test-group"
+        cc_res = self._create_consumer(group_id)
+        assert cc_res.status_code == requests.codes.ok
+
+        c0 = Consumer(cc_res.json(), self.logger)
+
+        # Subscribe a consumer to topic
+        self.logger.info(f"Subscribe consumer to topic: {self.topic}")
+        sc_res = c0.subscribe([self.topic])
+        assert sc_res.status_code == requests.codes.no_content
+
+        # Attempt to read from topic
+        self.logger.info(f"Consumer fetch compressed records")
+        cf_res = c0.fetch()
+        assert cf_res.status_code == requests.codes.ok, f"Status: {cf_res.status_code}"
+        result = cf_res.json()
+        self.logger.debug(f"fetch result {json.dumps(result)}")
+        for item in result:
```

Comment:
This success condition is quite loose:
- if the fetch returned zero records we wouldn't notice
- the offset check inside the loop could be used to assert the offsets count upwards as expected, rather than just being >-1